### DISCUSSION TITLE
How to sample 2x2 neighbor points of a given point, like  textureGather() in openGL

### DISCUSSION THREAD

**Discussion by zhouwangpei123** [halide/Halide#6852] [Q&A] [answered]
Given a point (x, y) in an image,  let intx = floor(x), inty = floor(y), can halide gather its four corner points (intx, inty), (intx+1, inty), (intx, inty+1), (intx+1, inty+1) in one time.

**Comment by abadams** [comment, marked as answer]
In Halide that has to be expressed as four separate loads.

**Reply by zhouwangpei123** [reply]
thanks for the clarification.

**Comment by soufianekhiat** [comment]
If I generate 4 loads dors Halide would be able to have intrinsic for gather4 on GPU?
